### PR TITLE
Add emscripten support

### PR DESCRIPTION
**Pull request by ByteDream**
This PR adds general support for the `*-emscripten` target.

- lua5.1 - lua5.4 ([lua-src-rs repo](https://github.com/khvzak/lua-src-rs)) already has lua support but this isn't published on crates.io yet (hence the git link in `Cargo.toml`)
- luau ([luau-src-rs repo](https://github.com/khvzak/luau-src-rs)) isn't supported yet but I have a open PR (khvzak/luau-src-rs#1)
- luajit doesn't support wasm and probably won't ever do so (https://github.com/LuaJIT/LuaJIT/issues/1112#issuecomment-1794370132)

### CODE REVIEW

File: mlua-sys/Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,6 @@ module = []
 cc = "1.0"
 cfg-if = "1.0"
 pkg-config = "0.3.17"
-lua-src = { version = ">= 546.0.0, < 546.1.0", optional = true }
+lua-src = { git = "https://github.com/khvzak/lua-src-rs", optional = true }
```
Comment (khvzak) [inline mlua-sys/Cargo.toml:41]: The new version with emscripten support is now available on crates.io

File: mlua-sys/src/lua51/compat.rs [hunk 2]
```diff
@@ -145,6 +145,12 @@ unsafe fn compat53_pushfuncname(L: *mut lua_State, ar: *mut lua_Debug) {
 // lua ported functions
 //
 
+#[inline(always)]
+pub unsafe fn lua_error(L: *mut lua_State) -> ! {
+    lua_error_(L);
+    unreachable!()
+}
```
Comment (khvzak) [inline mlua-sys/src/lua51/compat.rs:152]: It's not longer needed, see https://github.com/khvzak/mlua/commit/59974d7bdeaf7c1581198128e7ec275220664db0

**Comment by ByteDream** [comment]
I merged the latest master (with a bit of force pushing to keep my commits clean)

**Comment by ByteDream** [comment]
@khvzak I just found another function signature mismatch that I have overlooked (or I accidentally removed it when merging). The [`lua_sethook`](https://github.com/khvzak/mlua/blob/0b9a85e1838bed67ae69f11b42de84bcf19da80c/mlua-sys/src/lua52/lua.rs#L465) function in `lua52` returns `void` but it actually must be a `c_int`.

I didn't want to create a PR for a one-line change, hence the ping

**Pull request merged**